File: gm-utils/src/main/java/src/commom/utils/classe/ClassSimpleName.java
package src.commom.utils.classe;

import gm.languages.ts.javaToTs.JS;
import js.Js;
import src.commom.utils.string.StringCompare;

public class ClassSimpleName extends JS {

	public static String exec(Object o) {

		String s = Js.typeof(o);

		if (StringCompare.eqIgnoreCase(s, "string")) {
			return "string";
		}
		if (StringCompare.eq(s, "number")) {
			return "number";
		}

		if (Js.inJava) {
			if (o.getClass().equals(Double.class) || o.getClass().equals(Integer.class)) {
				return "number";
			}
			return o.getClass().getSimpleName();
		}

		return js(o).__proto__.constructor.name;

	}

	public static boolean is(Object o, String s) {
		return StringCompare.eq(exec(o), s);
	}

	public static boolean isObject(Object o) {
		return StringCompare.eq(exec(o), "Object");
	}

	public static boolean isString(Object o) {
		return StringCompare.eq(exec(o), "string");
	}

	public static boolean isNumber(Object o) {
		return StringCompare.eq(exec(o), "number");
	}

}

File: gm-utils/js/src/commom/utils/classe/ClassSimpleName.js
import StringCompare from '../string/StringCompare';

export default class ClassSimpleName {

	static exec(o) {

		let s = typeof(o);

		if (StringCompare.eqIgnoreCase(s, "string")) {
			return "string";
		}
		if (StringCompare.eq(s, "number")) {
			return "number";
		}

		let js = o;
		return js.__proto__.constructor.name;

	}

	static is(o, s) {
		return StringCompare.eq(ClassSimpleName.exec(o), s);
	}

	static isObject(o) {
		return StringCompare.eq(ClassSimpleName.exec(o), "Object");
	}

	static isString(o) {
		return StringCompare.eq(ClassSimpleName.exec(o), "string");
	}

	static isNumber(o) {
		return StringCompare.eq(ClassSimpleName.exec(o), "number");
	}

/* 	if (UNative.inJava) return !UType.isPrimitiva(o);
	return StringCompare.eq(Js.typeof(o), "object"); */

}
